fluent-syntax/src/parser.js

```diff
@@ -411,65 +409,170 @@ export default class FluentParser {
     return new AST.NumberLiteral(num);
   }
 
-  getValue(ps) {
-    if (ps.currentChar === "{") {
+  // getValue distinguished between patterns which start on the same line as the
+  // identifier (a.k.a. inline signleline patterns and inline multiline
+  // patterns) and patterns which start on a new line (a.k.a. block multiline
+  // patterns). The distinction is important for the dedentation logic: the
+  // indent of the first line of a block pattern must be taken into account when
+  // calculating the maximum common indent.
+  getValue(ps, {allowVariantList}) {
+    ps.peekBlankInline();
+    if (ps.isValueStart()) {
+      ps.skipToPeek();
+      return this.getValueType(
+        ps, {isBlock: false, allowVariantList});
+    }
+
+    ps.peekBlankBlock();
+    if (ps.isValueContinuation()) {
+      ps.skipToPeek();
+      return this.getValueType(ps, {isBlock: true, allowVariantList});
+    }
+
+    return null;
+  }
+
+  // Parse a VariantList (if allowed) or a Pattern.
+  getValueType(ps, {isBlock, allowVariantList}) {
```

Comment:
nit: since the `getValue` returns `Option<Value>` and `getValueType` returns `Value`, maybe we could rename it to `maybeGetValue` and `getValue`? I'm not super convinced that `maybe*` is the best naming scheme, but I don't like `getValueType` as it doesn't really convey the right difference from what `getValue` does.
Maybe you'll have a better idea. If not, it's ok to keep it imperfect :)

Reply:
I've been struggling with the naming of these methods and your comment suggests to me that my choices were not good enough :) Here's how the roles of these two methods differ:

- `getValue` distinguishes between inline and block values,
- `getValueType` distinguishes between `Patterns` and `VariantLists` (which are both value types).

Looking at this code today, I agree that this is confusing. Here's how I'm going to rename these methods before landing:

- `getValue` → `maybeGetValue`, to highlight the fact the it can return a `null`. It's the only `get*` method that does so. Thanks for the suggestion!
- `getValueType` → `getPatternOrVariantList`, to make its purpose clearer. _Value type_ was not immediately obvious.

Also note that when https://github.com/projectfluent/fluent/pull/220 lands, we'll be able to refactor this a bit and possibly remove `getPatternOrVariantList` because `VariantLists` will only be legal as values of `Terms`.